src/render/tutorials/graphql-usage/2_0.html.md

```diff
@@ -0,0 +1,785 @@
+        engine = new FragmentsEngine(vertx);
+    }
+
+    @Override
+    public CompletableFuture<T> get(DataFetchingEnvironment environment) {
+        FragmentEventContextTaskAware eventContextTaskAware = setupTask(vertx, config, routingContext, environment);
+        CompletableFuture<T> future = new CompletableFuture<>();
+
+        engine
+            .execute(Collections.singletonList(eventContextTaskAware))
+            .subscribe(events -> {
+                // Here we'll get the outcome fragment's payload, parse it to model and complete the future with it
+            });
+        
+        return future;
+    }
+
+    //other methods that we'll implement in a moment
+}
+```
+
+There's a couple of things to explain here. Firstly, our fetcher is abstract - it will have one abstract method for transforming payload into objects from our model (`Book`).
+Also, it's generic. That's because it doesn't matter for it what type will it return - it may be `Book`, it may be an array `Book[]` or it may be anything else.
+
+`TaskDataFetcher` implements `DataFetcher` (every data fetcher needs to) of type `CompletableFuture<T>`. Java implementation of GraphQL allows us to return futures instead of plain objects. If a data fetcher returns a future, GraphQL will asynchronously wait for it to complete. Only when all futures returned by data fetchers complete, the GraphQL will construct a response.
+
+Our abstract fetcher creates an instance of `FragmentsEngine`. The engine can execute task trees. Also, in the `get` method we create `FragmentEventContextTaskAware`. It's basicaly a task tree ready to be executed by the engine. Let's define the `setupTask` method that will take a configuration and parse it into a task tree. We'll also need one constant.
+
+```java
+private static final String FRAGMENT_TYPE = "graphql-data";
+
+//...
+
+private FragmentEventContextTaskAware setupTask(Vertx vertx, JsonObject config, RoutingContext routingContext, DataFetchingEnvironment env)  {
+    JsonObject fragmentConfig = new JsonObject();
+    fragmentConfig.put(FRAGMENT_TYPE, taskName);
+    fragmentConfig.put("gql", new JsonObject(env.getArguments()));
+
+    Fragment fragment = new Fragment(FRAGMENT_TYPE, fragmentConfig, "");
+
+    RequestContext requestContext = routingContext.get(RequestContext.KEY);
+    ClientRequest clientRequest = requestContext.getRequestEvent().getClientRequest();
+
+    FragmentEvent event = new FragmentEvent(fragment);
+    FragmentEventContext eventContext = new FragmentEventContext(event, clientRequest);
+
+    FragmentsHandlerOptions options = new FragmentsHandlerOptions(config);
+    ActionProvider proxyProvider = new ActionProvider(options.getAction(), supplyFactories(), vertx.getDelegate());
+    TaskBuilder taskBuilder = new TaskBuilder(FRAGMENT_TYPE, options.getTasks(), proxyProvider);
+
+    Task task = taskBuilder
+        .build(fragment)
+        .orElseThrow(() -> new NullPointerException("No task built"));

```

Comment:
Throwing NullPointerException by yourself? Seems to me that you should have your own ```IncorrectTaskInitializationException```.

Reply:
Yeah, it was a placeholder I forgot about. As by @tomaszmichalak suggestion, I've changed to `IllegalStateException`. 

I think this tutorial is big enough already, it's noted in the tutorial that a custom exception is a good idea, though